Src/VimWpf/Implementation/CommandMargin/CommandMarginUtil.cs

```diff
@@ -211,5 +209,61 @@ string ICommandMarginUtil.GetStatus(IVimBuffer vimBuffer)
         }
 
         #endregion
+
+        public static string GetShowCommandText(IVimBuffer vimBuffer)
+        {
+            if (vimBuffer.IncrementalSearch.InSearch)
+                return string.Empty;
+
+            switch (vimBuffer.ModeKind)
+            {
+                case ModeKind.Normal:
+                    return string.IsNullOrEmpty(vimBuffer.NormalMode.Command) ? string.Concat(vimBuffer.BufferedKeyInputs.Select(x => x.Char)) : vimBuffer.NormalMode.Command;
+                case ModeKind.SelectBlock:
+                    return GetVisualModeShowCommandText(vimBuffer, vimBuffer.VisualBlockMode);
+                case ModeKind.SelectCharacter:
+                    return GetVisualModeShowCommandText(vimBuffer, vimBuffer.VisualCharacterMode);
+                case ModeKind.SelectLine:
+                    return GetVisualModeShowCommandText(vimBuffer, vimBuffer.VisualLineMode);
+                case ModeKind.VisualCharacter:
+                case ModeKind.VisualBlock:
+                case ModeKind.VisualLine:
+                    return GetVisualModeShowCommandText(vimBuffer, (IVisualMode) vimBuffer.Mode);
+            }
+
+            return string.Empty;
+        }
+
+        private static string GetVisualModeShowCommandText(IVimBuffer vimBuffer, IVisualMode visualMode)
+        {
+            if (visualMode.CommandRunner.Inputs.Any())
+                return string.Concat(visualMode.CommandRunner.Inputs.Select(x => x.Char));

```

Comment:
It's a minor point, but my impression from the code base is that the preferred style is braces around even single statements. But what @jaredpar thinks is what really matters.